acceptance/lib/puppet/acceptance/environment_utils.rb

```diff
@@ -0,0 +1,44 @@
+module Puppet
+  module Acceptance
+    module EnvironmentUtils
+
+     # generate a random string of 6 letters and numbers.  NOT secure
+      def random_string
+        [*('a'..'z'),*('0'..'9')].shuffle[0,8].join
+      end
+      private :random_string
+
+      # if the first test to call this has changed the environmentpath, this will cause trouble
+      #   maybe not the best idea to memoize this?
+      def environmentpath
+        @@memoized_environmentpath ||= master.puppet['environmentpath']
+      end
+
+      # create a tmpdir to hold a temporary environment bound by puppet environment naming restrictions
+      # symbolically link environment into environmentpath
+      def mk_tmp_environment(environment)
+        # add the tmp_environment to a set to ensure no collisions
+        @@tmp_environment_set ||= Set.new
+        deadman = 100; loop_num = 0
+        while @@tmp_environment_set.include?(tmp_environment = environment.downcase + '_' + random_string) do
```

Comment:
beaker.tmpdir could/should be used here. It's not perfect because its permissions need adjusted after creating the dir; and I'm not sure it gives any guarantee against random string collisions; but it is a beaker method that is having its functionality repeated here.